src/bthome_ble/parser.py

```diff
@@ -435,9 +435,24 @@ def _parse_payload(self, payload: bytes, sw_version: int) -> bool:
             if obj_meas_type == 0:
                 last_packet_id = self.packet_id
                 new_packet_id = parse_uint(payload[obj_data_start:next_obj_start])
-                if new_packet_id == last_packet_id:
+                if (
+                    self.last_service_info
+                    and adv_time - self.last_service_info.time > 5
+                ):
+                    _LOGGER.debug(
+                        "Not filtering packet_id, new time - old time > 5. "
+                        "New time: %i, Old time: %i",
+                        adv_time,
+                        self.last_service_info.time,
+                    )
+                    break
+                elif (
+                    last_packet_id is not None
+                    and new_packet_id <= last_packet_id
+                    and last_packet_id - new_packet_id < 128
+                ):
```

Comment:
I'd make this a named function to better show intent and document with a docstring its handling the packet id wrapping

Reply:
I also moved the logic for comparing packet time to make it clear what are the conditions for dropping a packet